### PR TITLE
fix(selenium): apply Selenium 3 and 4 workaround for `switchTo()` when using Safari Driver

### PR DESCRIPTION
**Pull request by Zidious**
This is the backwards compatible Selenium 3 workaround instead of https://github.com/dequelabs/axe-core-maven-html/pull/392. We will try out best to switch to the about blank tab now, instead of assuming it is always the last handle in the list. 

Added test coverage to the file as it was missing. 

Closes: https://github.com/dequelabs/axe-core-maven-html/issues/391

### CODE REVIEW

File: selenium/pom.xml [hunk 1]
```diff
@@ -1,215 +1,223 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <artifactId>selenium</artifactId>
-  <version>4.8.0</version>
-  <packaging>jar</packaging>
-  <parent>
-    <groupId>com.deque.html.axe-core</groupId>
-    <artifactId>virtual-superpackage</artifactId>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (Zidious) [inline selenium/pom.xml:2]: Apologies - formatting

File: selenium/src/test/java/com/deque/html/axecore/selenium/WebDriverExtensionsTest.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  public void shouldThrowWhenUnableToDetermineWindowHandle() {
+    class MockedDriver extends ChromeDriver {
+      public MockedDriver(ChromeOptions chromeOptions) {
+        super(chromeOptions);
+      }
+
+      @Override
+      public Object executeScript(String script, Object... args) {
+        super.executeScript(script, args);
+        return super.executeScript(script, args);
```
Comment (straker) [inline selenium/src/test/java/com/deque/html/axecore/selenium/WebDriverExtensionsTest.java:136]: Is the purpose of this to call the open window twice so there are two new handles? A comment would be nice describing that if so as I had to think about this code for a bit.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -42,7 +42,10 @@ updates:
     ignore:
       # TODO: https://github.com/dequelabs/axe-core-maven-html/issues/378
       - dependency-name: "org.seleniumhq.selenium:selenium-java"
-        versions: [">=4.14.0"]
+        versions: [ ">=4.14.0" ]
+      # TODO: https://github.com/dequelabs/axe-core-maven-html/issues/393
+      - dependency-name: "org.mockito:mockito-core"
+          versions: [ ">=5.0.0" ]
```
Comment (straker) [inline .github/dependabot.yml:48]: This is outside the scope of the pr. Would be nice to remove and put in it's own pr.

**Review by straker — DISMISSED** [review]

File: .github/dependabot.yml (hunk 3 above)
Comment (Zidious) [reply .github/dependabot.yml:48]: Whoops my mistake - This can be removed along with the corresponding issue. I did not need need Mockito to mock the driver, just extending the base class and override to test functionality was sufficient!

File: selenium/src/test/java/com/deque/html/axecore/selenium/WebDriverExtensionsTest.java (hunk 2 above)
Comment (Zidious) [reply selenium/src/test/java/com/deque/html/axecore/selenium/WebDriverExtensionsTest.java:136]: Yes, sorry! Will add a comment, this is to simulate the case where another window gets created other than `about:blank`. An edge case, but a good one to cover and get a identifiable error message for.

File: selenium/pom.xml (hunk 1 above)
Comment (Zidious) [reply selenium/pom.xml:2]: No longer required

**Review by straker — APPROVED** [review]
